### PR TITLE
Arrow left and right in browser (internet/lan/favorites)

### LINKED ISSUE #9038: Switch between browser/lan/favorites with arrow key left and right
image

I have this shortcut in [term-ux](https://github.com/chillerbot/chillerbot-ux/blob/0939db3949fc8c76ebc67f565b069e7f9e2e28e7/README.md#term-ux-a-ncurses-terminal-client) and it feels quite natural there. I keep trying to press it in the ddnet client.

Any objections? Otherwise I would send a pr.

### PR DESCRIPTION
**Pull request by ChillerDragon**
Closed #9038

https://github.com/user-attachments/assets/3cea6a2a-d30e-45f7-b03f-a1e45dfa6517

## Checklist

- [x] Tested the change ingame
- [x] Provided screenshots if it is a visual change
- [x] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

**Review by Robyt3 — CHANGES_REQUESTED** [review]
This should also support switching between the favorite community tabs.

**Comment by ChillerDragon** [comment]
lol i did not know thats a thing

**Comment by ChillerDragon** [comment]
https://github.com/user-attachments/assets/216fd5bc-4b0e-4152-94b2-8d9b3ac8457e

File: src/game/client/components/menus.cpp [hunk 1]
```diff
@@ -660,6 +660,27 @@ void CMenus::RenderMenubar(CUIRect Box, IClient::EClientState ClientState)
 		}
 		GameClient()->m_Tooltips.DoToolTip(&s_FavoritesButton, &Button, Localize("Favorites"));
 
+		int MaxPage = (Box.w < BrowserButtonWidth) ? PAGE_FAVORITES : ((PAGE_FAVORITE_COMMUNITY_1 - 1) + (int)ServerBrowser()->FavoriteCommunities().size());
```
Comment (Robyt3) [inline src/game/client/components/menus.cpp:663]: The intent of the `Box.w < BrowserButtonWidth` condition is very unclear. Check `ServerBrowser()->FavoriteCommunities().empty()` instead.

**Review by Robyt3 — CHANGES_REQUESTED** [review]

File: src/game/client/components/menus.cpp (hunk 1 above)
Comment (ChillerDragon) [reply src/game/client/components/menus.cpp]: I now removed the entire condition. Tested it with max and min amount of community favorites worked fine.

**Review by Robyt3 — APPROVED** [review]

**Pull request merged**